engine/gamesys/src/dmsdk/gamesys/resources/res_texture.h

```diff
@@ -22,6 +22,8 @@ namespace dmGameSystem
     struct TextureResource
     {
         dmGraphics::HTexture m_Texture;
+        uint16_t             m_OriginalWidth;
+        uint16_t             m_OriginalHeight;
```

Comment:
Having these accessible per texture, is a small cost in my opinion, to avoid having more dependencies on the graphics context later on.